.github/workflows/build.yml

```diff
@@ -0,0 +1,82 @@
+name: Build
+
+on:
+  push:
+    ignore-branches: [main, future]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  # The Pulumi version doesn't matter much, it's just used for sanity testing
+  # but worth to update every once in a while
+  PULUMI_VERSION: "v3.46.0"
+  REPOSITORY_BASE_PATH: spaceliftbuild
+
+jobs:
+  build:
+    if: ${{ github.ref_type != 'tag' }}
+    name: 🚧 ${{ matrix.lang }} on ${{ matrix.arch }} 
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 2 # ECR has pretty aggressive rate limiting 😪
+      matrix:
+        arch: [amd64, arm64]
+        lang: [javascript, python, golang, dotnet]
+    
+    env:
+      BASE_IMAGE_NAME: ${{ (matrix.lang == 'javascript' || matrix.lang == 'golang') && 'runner-pulumi-base-alpine' || 'runner-pulumi-base-debian' }}
+      MAIN_IMAGE_FULL_NAME: runner-pulumi-${{ matrix.lang }}-${{ matrix.arch }}:${{ github.sha }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@main
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        if: matrix.arch == 'arm64'
+        with:
+          platforms: linux/arm64
+
+      # Building doesn't work with buildx here because of
+      # a stupid behavior:
+      # https://github.com/moby/moby/issues/42893#issuecomment-1241274246
```

Comment:
Random question - is it worth us using a multistage build instead and continuing to use buildx? Or will that not work in this situation?

Reply:
yeah I think this could be refactored, but in the future sometime. I just want to get this over with for now.

Reply:
No worries.